File: backend/src/test/java/info/gamewise/lor/videos/deckcodeextractor/DescriptionWithDeckLinkExtractorTest.java
package info.gamewise.lor.videos.deckcodeextractor;

import org.junit.jupiter.api.Test;

import java.util.Optional;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.api.Assertions.assertTrue;

class DescriptionWithDeckLinkExtractorTest {

    private final DeckCodeExtractor extractor = new DescriptionWithDeckLinkExtractor();

    @Test
    void mobaDecksCode() {
        final var deckCode = "CEBQCAYFCAFAGCICBEISGOCLJRHVIYABAECTCAICAMEVKWYA";

        final var description = """
            A brief title about the video
            
            https://lor.mobalytics.gg/decks/code/%s
            
            Thank you!
            Have a nice day.
        """.formatted(deckCode);

        Optional<String> deckCodeOpt = extractor.extract(description);

        assertTrue(deckCodeOpt.isPresent());
        assertEquals(deckCodeOpt.get(), deckCode);
    }

    @Test
    void mobaInvalidDecksCodeUrl() {
        final var deckCode = "CEBQCAYFCAFAGCICBEISGOCLJRHVIYABAECTCAICAMEVKWYA";

        final var description = """
            A brief title about the video
            
            https://lor.mobalytics.gg/decks/%s
            
            Thank you!
            Have a nice day.
        """.formatted(deckCode);

        Optional<String> deckCodeOpt = extractor.extract(description);

        assertTrue(deckCodeOpt.isEmpty());
    }

    @Test
    void onlyDeckCode() {
        final var deckCode = "CEBQCAYFCAFAGCICBEISGOCLJRHVIYABAECTCAICAMEVKWYA";

        final var description = """
            A brief title about the video
            
            Deck code: %s
            
            Thank you!
            Have a nice day.
        """.formatted(deckCode);

        Optional<String> deckCodeOpt = extractor.extract(description);

        assertTrue(deckCodeOpt.isEmpty());
    }

}

File: backend/src/test/kotlin/info/gamewise/lor/videos/deckcodeextractor/DescriptionWithDeckLinkExtractorTest.kt
package info.gamewise.lor.videos.deckcodeextractor

import org.junit.jupiter.api.Assertions.*
import org.junit.jupiter.api.Test

internal class DescriptionWithDeckLinkExtractorTest {
    private val extractor: DeckCodeExtractor = DescriptionWithDeckLinkExtractor()

    @Test
    fun mobaDecksCode() {
        val deckCode = "CEBQCAYFCAFAGCICBEISGOCLJRHVIYABAECTCAICAMEVKWYA"
        val description: String = """
            A brief title about the video
            
            https://lor.mobalytics.gg/decks/code/$deckCode
            
            Thank you!
            Have a nice day.
        """.trimIndent()

        val extractedDeckCode: String? = extractor.extract(description)
        assertNotNull(extractedDeckCode)
        assertEquals(deckCode, extractedDeckCode)
    }

    @Test
    fun mobaInvalidDecksCodeUrl() {
        val deckCode = "CEBQCAYFCAFAGCICBEISGOCLJRHVIYABAECTCAICAMEVKWYA"
        val description: String = """
            A brief title about the video
            
            https://lor.mobalytics.gg/decks/$deckCode
            
            Thank you!
            Have a nice day.
        """.trimIndent()
        val extractedDeckCode: String? = extractor.extract(description)
        assertNull(extractedDeckCode)
    }

    @Test
    fun onlyDeckCode() {
        val deckCode = "CEBQCAYFCAFAGCICBEISGOCLJRHVIYABAECTCAICAMEVKWYA"
        val description: String = """
            A brief title about the video
            
            Deck code: $deckCode
            
            Thank you!
            Have a nice day.
        """.trimIndent()
        val extractedDeckCode: String? = extractor.extract(description)
        assertNull(extractedDeckCode)
    }
}